### PR TITLE
[FINERACT-1678] fix get odlest cob closed loan query

### PR DESCRIPTION
**Pull request by taskain7**
## Description

Fix for the 'FIND_OLDEST_COB_PROCESSED_LOAN' DB query.

### CODE REVIEW

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/domain/LoanRepository.java [hunk 1]
```diff
@@ -83,7 +83,7 @@ public interface LoanRepository extends JpaRepository<Loan, Long>, JpaSpecificat
     String FIND_ID_BY_EXTERNAL_ID = "SELECT loan.id FROM Loan loan WHERE loan.externalId = :externalId";
 
     // should follow the logic of `FIND_ALL_NON_CLOSED_LOANS_BY_LAST_CLOSED_BUSINESS_DATE` query
-    String FIND_OLDEST_COB_PROCESSED_LOAN = "select loan.id, loan.lastClosedBusinessDate from Loan loan where loan.lastClosedBusinessDate = (select min(l.lastClosedBusinessDate) from Loan l where loan.loanStatus in (100,200,300,303,304) and l"
+    String FIND_OLDEST_COB_PROCESSED_LOAN = "select loan.id, loan.lastClosedBusinessDate from Loan loan where loan.lastClosedBusinessDate = (select min(l.lastClosedBusinessDate) from Loan l where l.loanStatus in (100,200,300,303,304) and l"
```
Comment (adamsaghy) [inline fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/domain/LoanRepository.java]: Both of the selects needs to be filtered by status, no?

Comment (taskain7) [reply fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/domain/LoanRepository.java]: that's correct.
fixed it.

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/domain/LoanRepository.java [hunk 2]
```diff
@@ -83,8 +83,8 @@ public interface LoanRepository extends JpaRepository<Loan, Long>, JpaSpecificat
     String FIND_ID_BY_EXTERNAL_ID = "SELECT loan.id FROM Loan loan WHERE loan.externalId = :externalId";
 
     // should follow the logic of `FIND_ALL_NON_CLOSED_LOANS_BY_LAST_CLOSED_BUSINESS_DATE` query
-    String FIND_OLDEST_COB_PROCESSED_LOAN = "select loan.id, loan.lastClosedBusinessDate from Loan loan where loan.lastClosedBusinessDate = (select min(l.lastClosedBusinessDate) from Loan l where loan.loanStatus in (100,200,300,303,304) and l"
-            + ".lastClosedBusinessDate <> " + ":cobBusinessDate)";
+    String FIND_OLDEST_COB_PROCESSED_LOAN = "select loan.id, loan.lastClosedBusinessDate from Loan loan where l.loanStatus in (100,200,300,303,304) and loan.lastClosedBusinessDate = (select min(l.lastClosedBusinessDate) from Loan l where l"
```
Comment (adamsaghy) [inline fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/domain/LoanRepository.java]: The first should be `loan.loanStatus`

**Review by adamsaghy — APPROVED** [review]
LGTM

**Pull request merged**